lambdas/pkgselect/index.py

```diff
@@ -59,203 +36,220 @@ def file_list_to_folder(df: pd.DataFrame, limit: int, offset: int) -> dict:
     top-level folder view (a special case of the s3-select
     lambda).
     """
-    if {'physical_key', 'logical_key', 'size'}.issubset(df.columns):
-        groups = df.groupby(df.logical_key.str.extract('([^/]+/?).*')[0], dropna=True)
+    if {"physical_key", "logical_key", "size"}.issubset(df.columns):
```

Comment:
could you reduce patch to substantial changes?

Reply:
i will, but it won't make much difference